parlai/mturk/webapp/run_mocks/mock_turk_agent.py

```diff
@@ -0,0 +1,311 @@
+#!/usr/bin/env python3
+
+# Copyright (c) 2017-present, Facebook, Inc.
+# All rights reserved.
+# This source code is licensed under the BSD-style license found in the
+# LICENSE file in the root directory of this source tree. An additional grant
+# of patent rights can be found in the PATENTS file in the same directory.
+
+import logging
+import time
+from queue import Queue
+
+from parlai.core.agents import Agent
+from parlai.mturk.core.agents import AssignState
+import parlai.mturk.core.shared_utils as shared_utils
+
+# Special act messages for failure states
+MTURK_DISCONNECT_MESSAGE = '[DISCONNECT]'  # Turker disconnected from conv
+TIMEOUT_MESSAGE = '[TIMEOUT]'  # the Turker did not respond but didn't return
+RETURN_MESSAGE = '[RETURNED]'  # the Turker returned the HIT
+
+
+class MockTurkAgent(Agent):
+    """Mock turk agent that can act in a parlai mturk world"""
+
+    # MTurkAgent Possible Statuses
+    ASSIGNMENT_NOT_DONE = 'NotDone'
+    ASSIGNMENT_DONE = 'Submitted'
+    ASSIGNMENT_APPROVED = 'Approved'
+    ASSIGNMENT_REJECTED = 'Rejected'
+
+    MTURK_DISCONNECT_MESSAGE = MTURK_DISCONNECT_MESSAGE
```

Comment:
is there a reason to redefine these here?

Reply:
It's so that external files that haven't imported the `MTurkAgent` file can pull them from the agent itself. I should really be pulling these from said `MTurkAgent` file though!